kinto-remote-settings/src/kinto_remote_settings/signer/__init__.py

```diff
@@ -277,3 +282,51 @@ def on_new_request(event):
         config.add_subscriber(send_notification, ReviewRejected)
     except ImportError:
         pass
+
+    # Automatically create resources on startup if option is enabled.
+    def auto_create_resources(event, resources):
+        storage = event.app.registry.storage
+        permission = event.app.registry.permission
+        write_principals = aslist(
+            event.app.registry.settings.get(
+                "signer.auto_create_resources_principals", []
+            )
+        ) or [Authenticated]
+
+        for resource in resources.values():
+            perms = {"write": write_principals}
+            bucket = resource["source"]["bucket"]
+            collection = resource["source"]["collection"]
+
+            bucket_uri = f"/buckets/{bucket}"
+            storage_create_raw(
+                storage_backend=storage,
+                permission_backend=permission,
+                resource_name="bucket",
+                parent_id="",
+                object_uri=bucket_uri,
+                object_id=bucket,
+                permissions=perms,
+            )
+
+            # If resource is configured for specific collection, create it too.
+            if collection:
+                collection_uri = bucket_uri + f"/collections/{collection}"
```

Comment:
nit: could include `bucket_uri` in the f-string

```suggestion
                collection_uri = f"{bucket_uri}/collections/{collection}"
```